src/registries/yarn-registry.js

```diff
@@ -72,18 +72,27 @@ export default class YarnRegistry extends NpmRegistry {
     if (typeof val === 'undefined') {
       val = DEFAULTS[key];
     }
-    
+
     return val;
   }
 
   async loadConfig(): Promise<void> {
-    for (const [isHome,, file] of await this.getPossibleConfigLocations('.yarnrc')) {
+    for (const [isHome, loc, file] of await this.getPossibleConfigLocations('.yarnrc')) {
       const config = parse(file);
 
       if (isHome) {
         this.homeConfig = config;
       }
 
+      // normalize offline mirror path relative to the current yarnrc
+      const offlineLoc = config['yarn-offline-mirror'];
+
+      // don't normalize if we already have a mirror path
+      if (!this.config['yarn-offline-mirror'] && offlineLoc) {
+        const mirrorLoc = config['yarn-offline-mirror'] = path.resolve(path.dirname(loc), offlineLoc);
+        await fs.mkdirp(mirrorLoc);
```

Comment:
Ah, I just missed that YarnRegistry is descendant from NpmRegistry.
My trouble is that loadConfig is not symmetrical between these 2 Registries, one of them creates mkdirp the other one does not.

Feels a bit fragile because it is not clear which of the two is loaded first.
Should we just add `await fs.mkdirp(mirrorLoc);` to both?